browser/components/BoardShowPage/MenuSideBar/ActivityPanel.js

```diff
@@ -2,20 +2,29 @@ import React, { Component } from 'react'
 import Link from '../../Link'
 import moment from 'moment'
 import './ActivityPanel.sass'
+import $ from 'jquery'
 
 class ActivityPanel extends Component {
 
   static PropTypes = {
     board: React.PropTypes.object.isRequired
   }
 
+  constructor(props) {
+    super(props)
+    this.state = {
+      refs: ''
+    }
+  }
+
   render() {
-    const activities = this.props.board.activity.map(activity =>
-      <Activity key={activity.id} activity={activity} users={this.props.board.users} board={this.props.board}/>
-    )
+    const activities = this.props.board.activity.map(activity => {
+      if ((this.props.card && this.props.card.id === activity.card_id) || !this.props.card) {
```

Comment:
consider using `Array#filter` followed by `Array#map` instead of returning undefined when this conditional is false.